server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,202 @@
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.RepositoriesMetadata;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.repositories.RepositoriesService;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * RemoteStore Repository Registration helper
+ */
+public class RemoteStoreRepositoryRegistrationHelper {
+
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.segment.repository.name";
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_TYPE_ATTRIBUTE_KEY = "remote_store.segment.repository.type";
+    public static final String REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY = "remote_store.segment.repository.settings";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.translog.repository.name";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY = "remote_store.translog.repository.type";
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY = "remote_store.translog.repository.settings";
+
+    private static void validateAttributeNonNull(DiscoveryNode joiningNode, String attributeKey) {
+        if (joiningNode.getAttributes().get(attributeKey) == null) {
+            throw new IllegalStateException("joining node [" + joiningNode + "] doesn't have the node attribute [" + attributeKey + "]");
+        }
+    }
+
+    /**
+     * A node will be declared as remote store node if it has any of the remote store node attributes.
+     * The method validates that the joining node has any of the remote store node attributes or not.
+     * @param joiningNode
+     * @return boolean value on the basis of remote store node attributes.
+     */
+    public static boolean isRemoteStoreNode(DiscoveryNode joiningNode) {
+        Map<String, String> joiningNodeAttributes = joiningNode.getAttributes();
+        boolean remoteStoreNode = joiningNodeAttributes.get(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY) != null
+            || joiningNodeAttributes.get(REMOTE_STORE_SEGMENT_REPOSITORY_TYPE_ATTRIBUTE_KEY) != null
+            || joiningNodeAttributes.get(REMOTE_STORE_SEGMENT_REPOSITORY_SETTINGS_ATTRIBUTE_KEY) != null
+            || joiningNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY) != null
+            || joiningNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_TYPE_ATTRIBUTE_KEY) != null
+            || joiningNodeAttributes.get(REMOTE_STORE_TRANSLOG_REPOSITORY_SETTINGS_ATTRIBUTE_KEY) != null;
+
+        if (remoteStoreNode) {
+            validateAttributeNonNull(joiningNode, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
```

Comment:
nit - we can have a set/list which has required settings that needs to be validated. And we can just iterate over the list reducing # of lines of code here.